test/plugins/iamauth.js

```diff
@@ -658,4 +607,56 @@ describe('#db IAMAuth Plugin', function() {
       });
     });
   });
+
+  it('returns an error if access token returns non-200 response', function(done) {
+    if (process.env.NOCK_OFF) {
+      this.skip();
+    }
+
+    var iamMocks = nock(TOKEN_SERVER)
+      .post('/identity/token', {
+        'grant_type': 'urn:ibm:params:oauth:grant-type:apikey',
+        'response_type': 'cloud_iam',
+        'apikey': IAM_API_KEY
+      })
+      .times(3)
+      .reply(500, 'Internal Error 500\nThe server encountered an unexpected condition which prevented it from fulfilling the request.');
+
+    var cloudantClient = new Client({ plugins: { iamauth: { iamApiKey: IAM_API_KEY } } });
+    var req = { url: SERVER + DBNAME, method: 'GET' };
+    cloudantClient.request(req, function(err) {
+      assert.equal(err.message, 'Failed to acquire access token. Status code: 500');
+      iamMocks.done();
+      done();
+    });
+  });
+
+  it('returns an error if IAM cookie login returns non-200 response', function(done) {
```

Comment:
This also has the same name as the earlier test (line 403), but I can see more differences than in the previous case. Perhaps we could rename slightly if they are testing different functionality.

Reply:
A dupe too I think. I'll remove.